File: Exercise List/2.py
def bubble_sort(l):
    num_swaps = 0
    for i in range(len(l)):
        swap = False
        for j in range(0, len(l) - 1 - i):
            if l[j] > l[j + 1]:
                l[j], l[j + 1] = l[j + 1], l[j]
                swap = True
                num_swaps += 1
        if not swap:
            return num_swaps
    return num_swaps

num_cases = int(input())

for i in range(num_cases):
    num_trains = int(input())
    trains = list(map(int, input().split()))
    print(f"Optimal train swapping takes {bubble_sort(trains)} swaps.")

File: Exercise List/2.cpp
#include <iostream>
using namespace std;

int bubble_sort(int trains[50], int num){
    int temp;
    int swaps = 0;
    for(int i = 0; i < num; i++) {
        for(int j = num-1; j > i; j--) {
            if(trains[j] < trains[j-1]) {
                temp = trains[j];
                trains[j] = trains[j-1];
                trains[j-1] = temp;
                swaps++;
            }
        }
    }
    return swaps;
}

int main() {
    int num_cases, num_trains, trains[50];
    cin>>num_cases;
    for(int _ = 0; _ < num_cases; _++) {
        
        cin>>num_trains;
        
        for(int i = 0; i < num_trains; i++)
           cin>>trains[i];
        
       cout<<"Optimal train swapping takes "<<bubble_sort(trains, num_trains)<<" swaps." <<endl;
    }
}